plone/app/content/tests/test_widgets.py

```diff
@@ -459,6 +473,8 @@ def testFileUpload(self):
         chunk.filename = 'test.xml'
         self.request.form['file'] = chunk
         self.request.REQUEST_METHOD = 'POST'
+        # the next calls plone.app.dexterity.factories and does a
+        # transaction.commit. Needs cleanup and FunctionalTesting layer.
```

Comment:
That's the usual problem yes, that in an integration layer there are ``transaction.commit()`` somewhere. The other usual problem is mixing ZopeTestCase with plone.app.testing tests.